### PR TITLE
[BUG] Sign transaction should allow unfunded signer

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
What
This improves the check for `hasEnoughXlm` during transaction signing in order to allow for an unfunded account to be the signer.

Why
In multi signature scenarios, this should be allowed given that the the signer may not be the tx source account.

Fixes: https://github.com/stellar/freighter/issues/1490

### CODE REVIEW

File: config/jest/setupTests.tsx [hunk 1]
```diff
@@ -35,13 +35,12 @@ jest.mock("react-i18next", () => ({
       changeLanguage: () => new Promise(() => {}),
     },
   }),
-  Trans: ({ children }: { children: React.ReactElement[] }) =>
-    children.map((child, i) => {
-      if (typeof child === "string") {
-        return <span key={i}>{child}</span>;
-      }
-      return "";
-    }),
```
Comment (piyalbasu) [inline config/jest/setupTests.tsx:44]: just curious - why was this necessary?

**Review by piyalbasu — APPROVED** [review]

File: config/jest/setupTests.tsx (hunk 1 above)
Comment (aristidesstaffieri) [reply config/jest/setupTests.tsx:44]: So it looks to me like this was incorrect but no test had hit this path to exercise it yet. In every case that we use Trans, we use a string as its child so expecting an array broke the test I was writing as it tried to render that warning.
